internal/decryption/output.go

```diff
@@ -21,39 +21,87 @@ import (
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-type AuditLogOutput struct {
-	Warningf  func(lineNb int, format string, a ...interface{}) error
-	Error     func(lineNb int, err error) error
-	Errorf    func(lineNb int, format string, a ...interface{}) error
-	LogRecord func(lineNb int, logRecord interface{}) error
-}
-
-func buildOutput(out io.Writer) AuditLogOutput {
-	newLine := []byte{'\n'}
-	writeLine := func(value []byte) error {
-		if _, err := out.Write(value); err != nil {
-			return err
-		}
-		_, err := out.Write(newLine)
+type AuditLogOutput interface {
+	Warningf(lineNb int, format string, a ...interface{}) error
+	Error(lineNb int, err error) error
+	Errorf(lineNb int, format string, a ...interface{}) error
+	LogRecord(lineNb int, logRecord interface{}) error
+}
+
+type auditLogOutputImpl struct {
+	out io.Writer
+}
+
+type AuditLogErrorLevel string
+
+const (
+	AuditLogErrorLevelError   AuditLogErrorLevel = "decryptionLogError"
+	AuditLogErrorLevelWarning AuditLogErrorLevel = "decryptionLogWarning"
+)
+
+type AuditLogError struct {
+	Level AuditLogErrorLevel
+	Line  int
+	Err   error
+}
+
+func (e AuditLogError) MarshalBSON() ([]byte, error) {
```

Comment:
is this used?

Reply:
yes by the bson marshaler

Reply:
https://github.com/mongodb/mongodb-atlas-cli/pull/1061/files#diff-f92e3e928b3bb2039c0c5949652bbfcff0a03b58998de03f746dee968b709da2R68

Reply:
it is actually an interface https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#Marshaler